### PR TITLE
[3.4] Add flag to pass initial guess for camera intrinsics into the calibration exe

### PR DESCRIPTION
**Pull request by catree**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [X] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

---

Allow passing some rough camera intrinsic parameters as an initial guess allow converging to the correct intrinsic parameters with my data.

With some `2048x2048` calibration images:
- with `CALIB_FIX_K3=1`, reprojection error is `1.2656`
- with `CALIB_FIX_K3=0`, reprojection error is `5.1693`
- with `CALIB_FIX_K3=0` and by using `fx=1400, fy=1400, cx=1024, cy=1024`, reprojection error is `0.39724`

The same images but resized to `1024x1024`:
- with `CALIB_FIX_K3=1`, reprojection error is `0.65225`
- with `CALIB_FIX_K3=0`, reprojection error is `2.31561`
- with `CALIB_FIX_K3=0` and by using `fx=700, fy=700, cx=512, cy=512`, reprojection error is `0.20709`

### CODE REVIEW

File: samples/cpp/calibration.cpp [hunk 1]
```diff
@@ -58,17 +56,24 @@ static void help(char** argv)
...
         "     [-oe]                    # write extrinsic parameters\n"
         "     [-zt]                    # assume zero tangential distortion\n"
-        "     [-a=<aspectRatio>]      # fix aspect ratio (fx/fy)\n"
+        "     [-a=<aspectRatio>]       # fix aspect ratio (fx/fy)\n"
         "     [-p]                     # fix the principal point at the center\n"
         "     [-v]                     # flip the captured images around the horizontal axis\n"
         "     [-V]                     # use a video file, and not an image list, uses\n"
         "                              # [input_data] string for the video file name\n"
         "     [-su]                    # show undistorted images after calibration\n"
+        "     [-ws=<number_of_pixel>]  # half of search window for cornerSubPix (11 by default)\n"
```
Comment (catree) [inline samples/cpp/calibration.cpp:70]: Parameter backported from 4.x.

File: samples/cpp/calibration.cpp [hunk 2]
```diff
@@ -154,8 +158,7 @@ static bool runCalibration( vector<vector<Point2f> > imagePoints,
     objectPoints.resize(imagePoints.size(),objectPoints[0]);
 
     double rms = calibrateCamera(objectPoints, imagePoints, imageSize, cameraMatrix,
-                    distCoeffs, rvecs, tvecs, flags|CALIB_FIX_K4|CALIB_FIX_K5);
-                    ///*|CALIB_FIX_K3*/|CALIB_FIX_K4|CALIB_FIX_K5);
+                    distCoeffs, rvecs, tvecs, flags | CALIB_USE_LU);
```
Comment (catree) [inline samples/cpp/calibration.cpp:161]: `CALIB_USE_LU` is used in OpenCV 4.x: https://github.com/opencv/opencv/blob/1fd45a1b8580ee90fa63c6106c78217c846247e4/samples/cpp/calibration.cpp#L173

Normally no need to pass `CALIB_FIX_K4|CALIB_FIX_K5` (`CALIB_RATIONAL_MODEL` should be explicitly passed), moreover `CALIB_FIX_K6` should be missing.

File: samples/cpp/calibration.cpp [hunk 3]
```diff
@@ -296,7 +298,7 @@ static bool runAndSave(const string& outputFilename,
     bool ok = runCalibration(imagePoints, imageSize, boardSize, patternType, squareSize,
                    aspectRatio, flags, cameraMatrix, distCoeffs,
                    rvecs, tvecs, reprojErrs, totalAvgErr);
-    printf("%s. avg reprojection error = %.2f\n",
+    printf("%s. avg reprojection error = %.7f\n",
```
Comment (catree) [inline samples/cpp/calibration.cpp:301]: Backport from 4.x: https://github.com/opencv/opencv/blob/1fd45a1b8580ee90fa63c6106c78217c846247e4/samples/cpp/calibration.cpp#L332

File: samples/cpp/calibration.cpp [hunk 4]
```diff
@@ -312,11 +314,10 @@ static bool runAndSave(const string& outputFilename,
     return ok;
 }
 
-
 int main( int argc, char** argv )
 {
     Size boardSize, imageSize;
-    float squareSize, aspectRatio;
+    float squareSize, aspectRatio = 1;
```
Comment (catree) [inline samples/cpp/calibration.cpp:320]: Backport from 4.x: https://github.com/opencv/opencv/blob/1fd45a1b8580ee90fa63c6106c78217c846247e4/samples/cpp/calibration.cpp#L353

File: samples/cpp/calibration.cpp [hunk 5]
```diff
@@ -362,12 +366,13 @@ int main( int argc, char** argv )
     }
     squareSize = parser.get<float>("s");
     nframes = parser.get<int>("n");
-    aspectRatio = parser.get<float>("a");
     delay = parser.get<int>("d");
     writePoints = parser.has("op");
     writeExtrinsics = parser.has("oe");
-    if (parser.has("a"))
+    if (parser.has("a")) {
```
Comment (catree) [inline samples/cpp/calibration.cpp:372]: Backport from 4.x: https://github.com/opencv/opencv/blob/1fd45a1b8580ee90fa63c6106c78217c846247e4/samples/cpp/calibration.cpp#L404-L407

File: samples/cpp/calibration.cpp [hunk 6]
```diff
@@ -381,6 +386,24 @@ int main( int argc, char** argv )
         cameraId = parser.get<int>("@input_data");
     else
         inputFilename = parser.get<string>("@input_data");
+    int winSize = parser.get<int>("ws");
```
Comment (catree) [inline samples/cpp/calibration.cpp:389]: Backport from 4.x: https://github.com/opencv/opencv/blob/1fd45a1b8580ee90fa63c6106c78217c846247e4/samples/cpp/calibration.cpp#L421

File: samples/cpp/calibration.cpp [hunk 7]
```diff
@@ -471,8 +494,8 @@ int main( int argc, char** argv )
         }
 
        // improve the found corners' coordinate accuracy
-        if( pattern == CHESSBOARD && found) cornerSubPix( viewGray, pointbuf, Size(11,11),
-            Size(-1,-1), TermCriteria( TermCriteria::EPS+TermCriteria::COUNT, 30, 0.1 ));
+        if( pattern == CHESSBOARD && found) cornerSubPix( viewGray, pointbuf, Size(winSize,winSize),
+            Size(-1,-1), TermCriteria( TermCriteria::EPS+TermCriteria::COUNT, 30, 0.0001 ));
```
Comment (catree) [inline samples/cpp/calibration.cpp:498]: Backport from 4.x: https://github.com/opencv/opencv/blob/1fd45a1b8580ee90fa63c6106c78217c846247e4/samples/cpp/calibration.cpp#L519

File: samples/cpp/calibration.cpp [hunk 8]
```diff
@@ -339,7 +340,10 @@ int main( int argc, char** argv )
 
     cv::CommandLineParser parser(argc, argv,
         "{help ||}{w||}{h||}{pt|chessboard|}{n|10|}{d|1000|}{s|1|}{o|out_camera_data.yml|}"
-        "{op||}{oe||}{zt||}{a|1|}{p||}{v||}{V||}{su||}"
+        "{op||}{oe||}{zt||}{a||}{p||}{v||}{V||}{su||}"
+        "{ws|11|}"
+        "{fx||}{fy||}{cx||}{cy||}"
+        "{imshow-scale|1|}{enable-k3|1|}"
```
Comment (catree) [inline samples/cpp/calibration.cpp:346]: In 3.4, by default `enable-k3=1` to keep the same behavior than before with: `flags|CALIB_FIX_K4|CALIB_FIX_K5`

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**